File: src/test/java/kitchenpos/application/fake/FakeProductRepository.java
package kitchenpos.application.fake;

import kitchenpos.domain.Product;
import kitchenpos.domain.ProductRepository;

import java.util.*;
import java.util.stream.Collectors;

public class FakeProductRepository implements ProductRepository {

    private final Map<UUID, Product> productMap = new HashMap<>();
    @Override
    public Product save(Product product) {
        UUID uuid = UUID.randomUUID();
        product.setId(uuid);
        productMap.put(uuid, product);
        return product;
    }

    @Override
    public Optional<Product> findById(UUID productId) {
        return Optional.ofNullable(productMap.get(productId));
    }

    @Override
    public List<Product> findAllByIdIn(List<UUID> ids) {
        return ids.stream()
                .map(productMap::get)
                .collect(Collectors.toList());
    }

    @Override
    public List<Product> findAll() {
        return new ArrayList<>(productMap.values());
    }
}

File: src/test/kotlin/kitchenpos/application/fake/FakeProductRepository.kt
package kitchenpos.application.fake

import kitchenpos.domain.Product
import kitchenpos.domain.ProductRepository
import java.util.*

class FakeProductRepository : ProductRepository {
    private val products = mutableMapOf<UUID, Product>()

    override fun save(entity: Product?): Product {
        products[entity!!.id] = entity
        return entity
    }

    override fun findById(id: UUID): Optional<Product> {
        return Optional.ofNullable(products[id])
    }

    override fun findAll(): MutableList<Product> {
        return products.values.toMutableList()
    }

    override fun findAllByIdIn(ids: MutableList<UUID>?): MutableList<Product> {
        return products.filter {
            it.key in ids!!
        }.map {
            it.value
        }.toMutableList()
    }
}
